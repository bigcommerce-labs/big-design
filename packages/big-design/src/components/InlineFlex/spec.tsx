import 'jest-styled-components';
import React from 'react';
import { render } from 'react-testing-library';

import { InlineFlex } from './index';

test('render inline flex', () => {
  const { container } = render(<InlineFlex>Flex</InlineFlex>);

  expect(container.firstChild).toMatchSnapshot();
});

test('has display flex', () => {
  const { container } = render(<InlineFlex>Flex</InlineFlex>);

  expect(container.firstChild).toHaveStyle('display: inline-flex');
});

test('forwards styles', () => {
  const { container } = render(
    <InlineFlex className="test" style={{ background: 'red' }}>
      Flex
    </InlineFlex>,
  );

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});

test('Flex Item forwards styles', () => {
  const { container } = render(
    <InlineFlex.Item className="test" style={{ background: 'red' }}>
      Flex
    </InlineFlex.Item>,
  );

  expect(container.getElementsByClassName('test').length).toBe(1);
  expect(container.firstChild).toHaveStyle('background: red');
});
