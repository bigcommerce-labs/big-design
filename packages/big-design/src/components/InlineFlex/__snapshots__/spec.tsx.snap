// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render inline flex 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

@media (min-width:720px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<span
  class="c0"
  direction="row"
  wrap="nowrap"
>
  Flex
</span>
`;
